src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/CodeActions/ComponentAccessibilityCodeActionProviderTest.cs

```diff
@@ -0,0 +1,154 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Test.Common;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.Language.Components;
+using Microsoft.AspNetCore.Razor.LanguageServer.CodeActions;
+using Microsoft.AspNetCore.Razor.LanguageServer.Common;
+using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+using Microsoft.VisualStudio.Editor.Razor;
+using Moq;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using Xunit;
+using Range = OmniSharp.Extensions.LanguageServer.Protocol.Models.Range;
+using System.Linq;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Test.CodeActions
+{
+    public class ComponentAccessibilityCodeActionProviderTest : LanguageServerTestBase
+    {
+        [Fact]
+        public async Task Handle_InvalidFileKind()
+        {
+            // Arrange
+            var documentPath = "c:/Test.razor";
+            var contents = "";
+            var request = new CodeActionParams()
+            {
+                TextDocument = new TextDocumentIdentifier(new Uri(documentPath)),
+                Range = new Range(),
+            };
+
+            var location = new SourceLocation(0, -1, -1);
+            var context = CreateRazorCodeActionContext(request, location, documentPath, contents, new SourceSpan(0, 0));
+            context.CodeDocument.SetFileKind(FileKinds.Legacy);
+
+            var provider = new ComponentAccessibilityCodeActionProvider(new DefaultTagHelperFactsService(), FilePathNormalizer);
+
+            // Act
+            var commandOrCodeActionContainer = await provider.ProvideAsync(context, default);
+
+            // Assert
+            Assert.Null(commandOrCodeActionContainer);
+        }
+
+        [Fact]
+        public async Task Handle_OutsideComponent()
```

Comment:
What's "OutsideComponent" mean?  I suggest  updating  to be  a  little more specific  as  to what  this  is testing.